Integration/LdapAuthIntegration.php

```diff
@@ -122,14 +126,24 @@ public function authCallback($settings = [], $parameters = [])
             $ldap = new LdapClient($hostname, $port, $ldapVersion, $ssl, $startTls);
 
             try {
-                $userKey = $settings['user_key'];
-                $query = $settings['user_query'];
+                if ($isactivedirectory) {
+                    $patterns = array();
+                    $patterns[0] = '/[,]/';
+                    $patterns[1] = '/DC=|dc=|d=/';
+                    $patterns[2] = '/\w{1,2}=\w+\./'; // delete all other common names and attributes
+                    $replacements = array();
+                    $replacements[0] = '.'; // the only actual replacement
+                    $parsed_dn = preg_replace($patterns, $replacements, $base_dn);
```

Comment:
I'm not really sure about this... that's a lot of intelligence and risks to compute (on every login) something that will never change. When I say "_risks_", I'm not an expert with AD, but what if the base DN does not only contain `dc` entries but also `ou` or some other things not handled by this ?
It might be better / more secure to add a `activedirectory_domain` property to `local.php`. This way, the admin is responsible of this setting this and it will improve login perf.

What do you think ? Is there any reason to compute this every time ?

Reply:
Mainly I wanted to have just one switch to turn AD of and on. Actually, I built the regex in a way that all the entries that are not prepend `DC=`/`dc=`/`d=` such as `o=users`, `Cn=sbs`, etc. would be deleted.

Nevertheless it will be much more efficient to introduce another parameter instead of computing something that never changes. 

I am going to change this asap, so you can merge the request then.